src/composables/trade/useGnosis.ts

```diff
@@ -296,7 +296,7 @@ export default function useGnosis({
 
               state.errors.priceExceedsBalance = bnum(
                 formatUnits(maximumInAmount, tokenIn.value.decimals)
-              ).gt(balanceFor(tokenIn.value));
+              ).gt(balanceFor(tokenIn.value.address));
```

Comment:
@garethfuller I've added a type for `balanceFor` - I think we should type all the other methods too, so we can count on TS to catch those :)